canonical-web-and-design/maas-ui
# feat(ui): add project selection step to Add LXD KVM form

## Done

- Added project selection step to Add LXD KVM form
- Split `AddLxd` into two separate forms: `AuthenticateForm` and `SelectProjectForm`

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and click Add KVM
- Authenticate to an LXD server (bolla LXD at `10.1.238.1:8443` will work)
- Check that projects are successfully fetched and you are taken to the next step
- Enter the name of an existing project in the "New project" field and check that an error shows on blur
- Enter a new project name and submit the form.
- Check that a new LXD pod is added with the details you entered
- Do the same thing but this time choose an existing project (note that I will be adding validation for existing pod-projects in a follow-up, so for now just choose a project that isn't already associated with a pod)

## Fixes

Fixes #2226
